### ISSUE TITLE
Support configuration of artifact server

### ISSUE THREAD

**Issue by GerrietReents** [btc-ag/service-idl#87]
Language specific artifact servers should be configurable, for example for maven pom files.

**Comment by sigiesec** [comment]
Please make this more specific. Which repository might be defined there? In particular for Maven, the preferred place to specify the repository is the user's settings.xml rather than the pom.xml (in particular when there is a maven resolver/mirror that should be used for all packages).

If you still feel, this should be supported for some target technology, please create technology-specific issues.

**Comment by GerrietReents** [comment]
Currently this is generated into the pom.xml file
```
<distributionManagement>
      <repository>
         <id>cab-maven</id>
         <name>CAB Main Maven Repository</name>
         <url>https://artifactory.bop-dev.de/artifactory/cab-maven/</url>
      </repository>
</distributionManagement>
```
As far as I understand, you can configure login credentials in settings.xml, but the url is definied in the pom.xml file:
From https://maven.apache.org/settings.html

> The repositories for download and deployment are defined by the repositories and distributionManagement elements of the POM. However, certain settings such as username and password should not be distributed along with the pom.xml.

**Comment by sigiesec** [comment]
Yes, this is done at the moment, but only because the maven environment on the Jenkins instance that runs the integration tests is not configured correctly. It ought to be removed.

For some rationale on why configuring repositories in the pom.xml is generally not a good idea, see https://blog.sonatype.com/2009/02/why-putting-repositories-in-your-poms-is-a-bad-idea/

There might be exceptions. Which repository would you want to configure there?

**Comment by GerrietReents** [comment]
I was thinking about the artifact server for uploading the build results not the ones holding the dependencies. It is not possible to specify the distributionManagement element in settings.xml.
A solution would be to use a compony wide pom.xml like described here https://stackoverflow.com/questions/3298135/how-to-specify-mavens-distributionmanagement-organisation-wide?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

But than again, the name, id and version of that parent pom file has to be configured for the generator.

**Comment by sigiesec** [comment]
Ok, that's a different issue :)

Currently only module-specific pom.xml files are generated, which is probably not the right place. When we add generation of a parent pom.xml, this might be included there. However, the existing use cases seem to provide a handmade parent pom.xml, which contains not only the generated modules. For the integration tests, it makes sense to generate a parent pom.xml, but there we don't need any distributionManagement.

For a use case, which wants to make a release unit that exactly consists of the set of generated modules, this might make sense.

**Comment by GerrietReents** [comment]
Maybe we should change our approach for generating pom files (or any project files). These files seem to be very specific to the projects using the generator. Would it be better to let the user provide a template pom file with some replacement variables in it, that get changed by the generator? We could use something like the velocity template engine.

**Comment by sigiesec** [comment]
See also discussion in http://maven.40175.n5.nabble.com/Can-t-specify-distributionManagement-in-settings-xml-td3181781.html

**Comment by sigiesec** [comment]
@GerrietReents For our internal uses, I think this is not relevant anymore, when we only use the Jenkins and bop-pipelines macros. Of course, when this is going to be used externally, this is necessary. But I think we can move this to after 1.0.0. Please object if you see this differently.
